plugins/themes/controllers/themes_controller.py

```diff
@@ -451,6 +453,9 @@ def create_form(self, theme=None):
                 form.additionalMouseCrs.data = theme["additionalMouseCrs"]
             if "searchProviders" in theme:
                 form.searchProviders.data = theme["searchProviders"]
+                qgis_search = [provider for provider in theme["searchProviders"] if "provider" in provider and isinstance(provider, dict)]
+                if qgis_search :
+                    form.qgisSearchProvider.data = qgis_search[0]
```

Comment:
Is my reading correct that this will only match the first provider which is a dictionary entry? Note in particular that solr also is configured as a dictionary entry, and that you can potentially have multiple qgis searches configured.

Reply:
@manisandro Indeed, i've forgottent the usecase where you have multiple "provider" (solr for example).
It should works now. 

The condition check that if a provider is of type dict and got "qgis" it returns the element otherwise the list is empty and got filtered by 
"if qgis_search :".

Feel free to test it more.

Reply:
`[provider for provider in theme["searchProviders"] if isinstance(provider, dict) and provider.get("provider") == "qgis"]` 

is probably sufficient

But this will still only read the first configuration (`form.qgisSearchProvider.data = qgis_search[0]`)?

Reply:
Yes but as I understand there is only 1 Qgis search provider by theme ?
So we only need to return the first element which got filtered by the for loop ? 
And otherwise pass with the if.

Maybe I've missed something ?

Reply:
No, you can have an arbitrary number of qgis search providers per theme.

Reply:
@manisandro, Oh well i miss that indeed. 
Then, it complicate the things quite  bit because returning multiple json items in a single text editor is quite handy a not very confortable for the end user, especially since json.dumps seems to escape all the double quotes.

Reply:
@manisandro, Oh well, I miss that indeed.
Then, it complicate the things quite bit because returning multiple json items in a single text editor is quite handy and not very confortable for the end user, especially since json.dumps seems to escape all the double quotes.

I've looked at different solutions but I do not find one that would be comfortable in term of both code and user experience for now. 
Imbriquing a new subform to the existing form will result in too much clutter I think. Same for a modal form.

And making a separate form like the info_template doesn't seems very convenient either since the searchproviders are handled in the themesconfig part of the tenant config.

I will continue to look into it, because I think it's too bad to miss this feature, but if you have any solution to help me here I'll take it ! 

I was thinking maybe storing the Qgis search configs in another part of the tenantconfig.json with a dedicated form, then retrieve them conveniently in the theme could be easier ? A bit like what is done for the backgroundlayers 

Thanks,
Regards, Clément.

Reply:
> I was thinking maybe storing the Qgis search configs in another part of the tenantconfig.json with a dedicated form, then retrieve them conveniently in the theme could be easier ? A bit like what is done for the backgroundlayers

This works for background layers because they are completely theme independent, whereas a QGIS search is generally very tightly coupled to the layers of the respective theme. So it is not practical to store them outside the theme. 

I can't think of a solution other than exposing the complexity in the UI. A super generic approach would be to have the entire UI generated by a JSON schema definition.